cpp/src/arrow/compute/function_test.cc

```diff
@@ -307,7 +310,7 @@ TEST(ScalarAggregateFunction, Basics) {
 }
 
 Result<std::unique_ptr<KernelState>> NoopInit(KernelContext*, const KernelInitArgs&) {
-  return nullptr;
+  return NULLPTR;
```

Comment:
The `NULLPTR` macro is only useful in header files to avoid weird issues with Microsoft's C++/CLI, AFAIU.